ui/apps/platform/src/Containers/NetworkGraph/NetworkGraph.tsx

```diff
@@ -224,13 +200,7 @@ const TopologyComponent = ({ detailType, detailId }: TopologyComponentProps) =>
             // @ts-ignore
             const [newDetailType, newDetailId] = getUrlParamsForEntity(newSelectedEntity);
 
-            // if found, and it's not the logical grouping of all external sources, then trigger URL update
-            if (newSelectedEntity && newSelectedEntity?.data?.entityType !== 'EXTERNAL') {
-                history.push(`${networkBasePathPF}/${newDetailType}/${newDetailId}`);
-            } else {
-                // otherwise, return to the graph-only state
-                history.push(`${networkBasePathPF}`);
-            }
+            onSelectNode(newDetailType, newDetailId);
```

Comment:
I was thinking we don't need to do the URL manipulation on this level. We could have the NetworkGraphPage get/set the URL data and NetworkGraph can get/set the topology model data

Reply:
We're not setting the URL to _reflect_ the data that NetworkGraph component decides to display separately. The data should only change _after_ we update the URL. (Think of the URL as Redux, and a URL update as an "action".)

With that said, I think this change would be necessary once we add the filtering, so we should go ahead and move the URL handling up.

Reply:
Oh, yea you're right. I didn't go all the way. Let me make a few extra changes then